api/processors/actions/student/transfer.py

```diff
@@ -0,0 +1,50 @@
+from django.db import transaction
+from django.db.models import QuerySet
+
+from api.models import Group, Student
+from api.models.choices.log_event_type import StudentLogEventType
+from api.models.choices.status.project import StudentProjectStatus
+from api.models.log_event import StudentLogEvent
+from api.processors.actions.student import StudentActionProcessor
+
+
+class StudentTransferProcessor(StudentActionProcessor):
+    def __init__(self, student: Student, to_group: Group):
+        self.to_group = to_group
+        super().__init__(student)
+
+    @transaction.atomic
+    def process(self) -> None:
+        self._transfer()
+        self._set_statuses()
+        self._create_log_events()
+
+    def _transfer(self) -> None:
+        self.old_groups: QuerySet[Group] = self.student.groups.all()
+        for old_group in self.old_groups:
+            self.student.groups.remove(old_group)
+            old_group.students_former.add(self.student)
```

Comment:
hmm, this may, of course, be the intended behaviour, but personally, i'd rather ask for a `from_group` or smth in the request body and only remove the student from that group.

what would happen when a student is, for example, studying english A1 in one group and spanish A1 in another and is then transferred (upon their request) to an english B1 group?

Reply:
oops, had the same exact comment as @lemontree210 , lets continue in his thread 👌